src/PrestaShopBundle/Bridge/AdminController/LegacyControllerBridge.php

```diff
@@ -305,20 +343,43 @@ public function __get(string $name)
      * to allow legacy code setting properties in configuration as if it would be in legacy controller.
      *
      * @param string $name
-     * @param $value
+     * @param mixed $value
      *
      * @return void
      */
     public function __set(string $name, $value): void
     {
-        if (!isset($this->propertiesMap[$name])) {
-            $this->{$name} = $value;
+        $modifiedField = &$this->getPropertyReference($name);
+        $modifiedField = $value;
+    }
 
-            return;
+    /**
+     * We need to return a reference in order for legacy codes to do manipulation like:
+     *    $controller->toolbar_button['new_product'] = ['text' => 'New Product'];
+     *
+     * That's why we can't use a PropertyAccessor component because it does not return references, only values.
+     *
+     * @param string $name
+     *
+     * @return mixed
+     *
+     * @throws InvalidArgumentException
+     */
+    private function &getPropertyReference(string $name)
```

Comment:
interesting :mage_man:

Reply:
Yeah I couldn't find a solution that worked without this reference trick, and since the PropertyAccessor returns only values and not references we have to implement this manually

Although this could be improved later by relying on `PropertyPath` (to handle the parsing of the mapping) and we could add management for methods (automatic getters/setters) This means re-implementing the PropertyAccessor a bit but we don't have much choice here.

For now this should be enough, but in a future refacto when we change the configuration class and rely on getters/setters instead this will become a must-have to handle methods at least.